server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java

```diff
@@ -162,6 +165,7 @@ protected void doExecute(Task task, BulkRequest bulkRequest, ActionListener<Bulk
         final Metadata metadata = clusterService.state().getMetadata();
         final Version minNodeVersion = clusterService.state().getNodes().getMinNodeVersion();
         for (DocWriteRequest<?> actionRequest : bulkRequest.requests) {
+            prohibitAppendOnlyWritesInBackingIndices(actionRequest, metadata);
```

Comment:
I think the exception thrown will terminate the entire bulk request. Normally, only errors around the bulk structure itself will cause the entire bulk request to fail. Individual item problems are reported back per item instead. I would think this should be per item too.

I think moving the check into [here in `BulkOperation`](https://github.com/elastic/elasticsearch/blob/4f2f0b5ace82681d9a8f8cd9c457d929f15cd29c/server/src/main/java/org/elasticsearch/action/bulk/TransportBulkAction.java#L457) would fix this.